spacy_loggers/mlflow.py

```diff
@@ -66,23 +83,34 @@ def log_step(info: Optional[Dict[str, Any]]):
                 other_scores = info["other_scores"]
                 losses = info["losses"]
                 output_path = info.get("output_path", None)
+                if log_latest_dir:
+                    latest_model.update(output_path)
+                
                 if score is not None:
-                    mlflow.log_metric("score", score)
+                    mlflow.log_metric("score", score, info["step"])
                 if losses:
-                    mlflow.log_metrics({f"loss_{k}": v for k, v in losses.items()})
+                    mlflow.log_metrics({f"loss_{k}": v for k, v in losses.items()}, info["step"])
                 if isinstance(other_scores, dict):
                     mlflow.log_metrics(
                         {
                             k: v
                             for k, v in util.dict_to_dot(other_scores).items()
                             if isinstance(v, float) or isinstance(v, int)
-                        }
+                        },
+                        info["step"]
                     )
                 if output_path and score == max(info["checkpoints"])[0]:
                     nlp = load(output_path)
                     mlflow.spacy.log_model(nlp, "best")
+                    log_model(output_path, 'model_best')
```

Comment:
It irks me that now we have three artifacts - `best`, `model_best` and `model_last`. Ideally I'd like to have `best` and `last`, both the complete set of associated data (like `best` has, see screenshot). Just for the sake of consistency. If that doesn't work out due to the issues you described, I guess this is fine. I'll look into this.
![image](https://user-images.githubusercontent.com/7152225/206166793-6894b400-608f-45a2-ba4c-1760fa3900d9.png)

Reply:
> Note: I ended up using mlflow.log_artifact() rather than mlflow.spacy.log_model() because when I use the latter method outside of the log_step() function it looks like spacy just aborts the process abruptly, without letting the mlflow.spacy function do its thing.

Hm, I ran this with 
```python
        def finalize() -> None:
            if log_latest_dir:
                mlflow.spacy.log_model(nlp, "model_last")
```
now and this seemed to work (see image). What's your outcome?
![image](https://user-images.githubusercontent.com/7152225/206171066-29b93dc4-3f3d-444e-b3f3-eea666ddf34d.png)